### PR TITLE
Chore/add private protected types

### PR DESCRIPTION
**Pull request by Knorgias**
## What I did

1. Added @protected & @private to _ & __ properties and methods

### CODE REVIEW

File: packages/calendar/src/LionCalendar.js [hunk 1]
```diff
@@ -130,11 +130,13 @@ export class LionCalendar extends LocalizeMixin(LitElement) {
 
       /**
        * The currently focused date (if any)
+       * @private
```
Comment (jorenbroekema) [inline packages/calendar/src/LionCalendar.js:133]: I don't believe TS looks at static get properties for class fields, so these @private type annotations probably need to be moved to where the class field is initialised (usually in the constructor).

File: packages/combobox/src/LionCombobox.js [hunk 2]
```diff
@@ -253,22 +266,36 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
...
     this._ariaVersion = browserDetection.isChromium ? '1.1' : '1.0';
 
     /**
      * @configure ListboxMixin
+     * @protected
      */
     this._listboxReceivesNoFocus = true;
 
+    /**
+     * @protected
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:280]: private right?

File: packages/combobox/src/LionCombobox.js [hunk 3]
```diff
@@ -253,22 +266,36 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
...
+     */
     this.__prevCboxValueNonSelected = '';
+    /**
+     * @protected
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:284]: private

File: packages/combobox/src/LionCombobox.js [hunk 4]
```diff
@@ -523,6 +562,7 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
    * - complete: completes the textbox value inline (the 'missing characters' will be added as
    * selected text)
    *
+   * @private
```
Comment (jorenbroekema) [inline packages/combobox/src/LionCombobox.js:565]: protected

File: packages/form-core/src/registration/FormRegistrarMixin.js [hunk 5]
```diff
@@ -38,6 +38,7 @@ const FormRegistrarMixinImplementation = superclass =>
          * it should act like an array (see ChoiceGroupMixin).
          * Usually, when false, we deal with a choice-group (radio-group, checkbox-group,
          * (multi)select)
+         * @protected
```
Comment (jorenbroekema) [inline packages/form-core/src/registration/FormRegistrarMixin.js:41]: static get prop so this @protected won't do much, annotate in initialisation instead.

File: packages/form-core/src/validate/ValidateMixin.js [hunk 6]
```diff
@@ -83,6 +83,7 @@ export const ValidateMixinImplementation = superclass =>
         /**
          * Subclassers can enable this to show multiple feedback messages at the same time
          * By default, just like the platform, only one message (with highest prio) is visible.
+         * @protected
```
Comment (jorenbroekema) [inline packages/form-core/src/validate/ValidateMixin.js:86]: static get prop, see above comment

**Review by jorenbroekema — CHANGES_REQUESTED** [review]

**Comment by jorenbroekema** [comment]
Reviewed all commits since my previous review. Looks good 👍 

Now we just need to:
- Rebase on latest master --> fix merge conflict
- Add a changeset (patch) for all changed packages
- Squash all commits into one

Then we can merge :)

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 7]
```diff
@@ -145,12 +158,15 @@ describe('<lion-checkbox-indeterminate>', () => {
...
     elIndeterminate._inputNode.click();
     await elIndeterminate.updateComplete;
+    const { subCheckboxes } = getProtectedMembers(elIndeterminate);
 
     // Assert
-    expect(elIndeterminate?.hasAttribute('indeterminate')).to.be.false;
-    expect(elIndeterminate?._subCheckboxes[0].hasAttribute('checked')).to.be.true;
-    expect(elIndeterminate?._subCheckboxes[1].hasAttribute('checked')).to.be.true;
-    expect(elIndeterminate?._subCheckboxes[2].hasAttribute('checked')).to.be.true;
+    if (elIndeterminate) {
```
Comment (tlouisse) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:164]: I think the test should fail if `elIndeterminate` is null (is it possible to remove this if? (according to lin 154, TS shouldn't complain)

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 8]
```diff
@@ -251,13 +273,21 @@ describe('<lion-checkbox-indeterminate>', () => {
...
+
     // Assert - the second sibling should not be affected
-    expect(elFirstIndeterminate?.hasAttribute('indeterminate')).to.be.false;
-    expect(elFirstIndeterminate?._subCheckboxes[0].hasAttribute('checked')).to.be.true;
-    expect(elFirstIndeterminate?._subCheckboxes[1].hasAttribute('checked')).to.be.true;
-    expect(elFirstIndeterminate?._subCheckboxes[2].hasAttribute('checked')).to.be.true;
-    expect(elSecondIndeterminate?._subCheckboxes[0].hasAttribute('checked')).to.be.false;
-    expect(elSecondIndeterminate?._subCheckboxes[1].hasAttribute('checked')).to.be.false;
+
+    if (elFirstIndeterminate) {
```
Comment (tlouisse) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:281]: see comments above

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 9]
```diff
@@ -251,13 +273,21 @@ describe('<lion-checkbox-indeterminate>', () => {
...
+      expect(elFirstIndeterminate.hasAttribute('indeterminate')).to.be.false;
+      expect(elFirstSubCheckboxes.subCheckboxes[0].hasAttribute('checked')).to.be.true;
+      expect(elFirstSubCheckboxes.subCheckboxes[1].hasAttribute('checked')).to.be.true;
+      expect(elFirstSubCheckboxes.subCheckboxes[2].hasAttribute('checked')).to.be.true;
+    }
+    if (elSecondIndeterminate) {
```
Comment (tlouisse) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:287]: see comments above

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 10]
```diff
@@ -301,18 +331,20 @@ describe('<lion-checkbox-indeterminate>', () => {
     const elParentIndeterminate = /**  @type {LionCheckboxIndeterminate} */ (el.querySelector(
       '#parent-checkbox-indeterminate',
     ));
+    const ElNestedSubCheckboxes = getProtectedMembers(elNestedIndeterminate);
```
Comment (tlouisse) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:334]: Instance or 'regular variable' names should not be capitalized. So usually we have the convention:

```js
class MyClass {}

const myClass = new MyClass();

```

So in the example above `MyClass` is the constructor, `myClass` is the instance. As a rule of thumb, we can say that only constructor names should be capitalized, all other variable names should start with a lowercase char.

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 11]
```diff
@@ -301,18 +331,20 @@ describe('<lion-checkbox-indeterminate>', () => {
...
+    const ElParentSubCheckboxes = getProtectedMembers(elParentIndeterminate);
 
     // Act - check a nested checkbox
-    elNestedIndeterminate?._subCheckboxes[0]._inputNode.click();
+    if (elNestedIndeterminate) ElNestedSubCheckboxes.subCheckboxes[0]._inputNode.click();
```
Comment (tlouisse) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:338]: We have the convention to always use parentheses and newlines for an if for readability (should be captured by linting though...)

**Comment by tlouisse** [comment]
Hi, great start 💪 

If we look at [this](https://github.com/ing-bank/lion/issues/1140) issue, not everything is tackled. (we focused mainly on point 2 now, I think.)
Next step would be to also add all the info to the mixin type declarations and add all the descriptions there as well (especially for public methods and protected methods)

**Review by tlouisse — APPROVED** [review]

**Pull request merged**